Tests/base/NSRunLoop/dispatch.m

```diff
@@ -8,14 +8,12 @@
 
 const NSTimeInterval kDelay = 0.01;
 
-#if HAVE_LIBDISPATCH_RUNLOOP && __has_feature(blocks)
+#if GS_USE_LIBDISPATCH_RUNLOOP && __has_feature(blocks)
```

Comment:
In the implementation, we only need to call C functions from libdispatch, but in the test, we also need to schedule blocks onto the main queue to verify that running the runloop will drain the main queue as well. Hence the subtle difference: The test code depends on a blocks capable compiler, the implementation code doesn't.

Reply:
Sorry, I was rather aiming at the lines below, not that one by itself.

Reply:
Ah, got it: The `HAVE_DISPATCH_H` defines etc. come from `Source/config.h`, so it doesn't seem like they were supposed to be exposed to the test framework (otherwise it should be in a `Headers/` subdirectorry -- probably one of the reasons why this code path broke down with time). And since this is effectively a clang only code path, it made sense to do it this way.

This is very defensive anyways: I haven't a build of libdispatch that didn't place its headers in subdirectory for a long time.